homeassistant/components/homekit/type_fans.py

```diff
@@ -83,6 +98,56 @@ def set_oscillating(self, value):
                   ATTR_OSCILLATING: oscillating}
         self.call_service(DOMAIN, SERVICE_OSCILLATE, params, oscillating)
 
+    def set_speed_list(self, speed_list):
```

Comment:
We don't need to save the complete speed list inside the accessory. After the check the `speed_list_type` should be enough for the conversion later.

Reply:
Still new to Home Assistant, I thought that maybe this should support later alterations of speed_list via the Customization UI; but I do agree that assuming that the list stays the same makes the code simpler and easier to understand.